packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue

```diff
@@ -1,20 +1,44 @@
 <script lang="ts" setup>
+import graphSearch from "@sit-onyx/icons/graph-search.svg?raw";
+import { computed, type VNode } from "vue";
 import { useDensity } from "../../composables/density";
+import { injectI18n } from "../../i18n";
+import OnyxEmpty from "../OnyxEmpty/OnyxEmpty.vue";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
 import type { OnyxTableProps } from "./types";
 
 const props = withDefaults(defineProps<OnyxTableProps>(), {
   striped: false,
   withVerticalBorders: false,
 });
 
-defineSlots<{
+const slots = defineSlots<{
   /**
    * Table content. Must only contain valid HTML `<table>` children like `<tr>`, `<th>`, `<td>`, `<thead>` and `<tbody>`.
    */
-  default(): unknown;
+  default(): VNode[];
+  /**
+   * Optional slot to customize the empty state when no body content exist.
+   * It is recommended to use the `<OnyxEmpty>` component here.
+   *
+   * If unset, a default translated message will be displayed for the current locale.
+   */
+  empty?(props: { defaultMessage: string }): unknown;
 }>();
 
+const { t } = injectI18n();
+
 const { densityClass } = useDensity(props);
+
+/**
+ * Returns an empty message if no table body was found
+ * or no rows were found inside a table body.
+ */
+const isEmptyMessage = computed(() => {
+  const tableBody = slots.default?.().find((data) => data.type === "tbody");
```

Comment:
I am not sure if we should do this since it can be flaky / have edge cases.
Jonathan mentioned that he doesn't like to iterate over slots.

We could implement it via plain CSS and just set the empty to `display: none`

Reply:
> if we should do this

what do you mean -> 
1) look into our slot to find out if we should show an empty text
2) use v-if on the onyx-table__empty tbody?

I don't see the connection of your first paragraph and the last sentence ^^

Reply:
if we aren't allowed to look into the slots.default, this means either

1) we are forced to split the default slot into a "thead" and a "tbody" slot
or
2) we need to add a prop to determine if we should show "empty", and ignore what's in the slot

Reply:
Plain CSS solution could look something like this:

```css
.onyx-table {
    &__empty {
      display: none;
    }

    &:not(:has(tbody tr)) {
      .onyx-table__empty {
        display: block;
      }
    }
}
```

Reply:
> > if we should do this
> 
> what do you mean ->
> 
> 1. look into our slot to find out if we should show an empty text
> 2. use v-if on the onyx-table__empty tbody?
> 
> I don't see the connection of your first paragraph and the last sentence ^^

I meant I am not sure if we should look into the slot content and search for a tbody element

Reply:
Example why this can be flaky:

- we can not force the project to pass in a `tbody`. You can also just pass `<tr>` which is valid
- HTML will insert a tbody automatically when its rendered
- so if we filter the slot content, there will be no tbody element, although there is table data available
- when checking for `:has(tbody tr)` with CSS to check if the table is empty, this will be always correct because since HTML will add a `tbody` if you don't provide one

Reply:
ok, got it 👍

Reply:
I'll have to refactor that onyx-table__empty itself isn't a tbody then. ^^ else the condition never is fulfilled

Reply:
> I'll have to refactor that onyx-table__empty itself isn't a tbody then. ^^ else the condition never is fulfilled

Either so or we could change the selector to:

```css
&:not(:has(tbody tr:not(:has(.onyx-table__empty))))
```

But thats a lot of :not and :has :D